src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -333,6 +345,167 @@ DatabaseShardImp::validate()
     app_.shardFamily()->reset();
 }
 
+void
+DatabaseShardImp::importNodeStore()
+{
+    std::unique_lock<std::mutex> l(m_);
+    assert(init_);
+    std::uint32_t earliestIndex;
+    std::uint32_t latestIndex;
+    {
+        uint256 hash;
+        std::uint32_t seq;
+        std::shared_ptr<Ledger> ledger;
+
+        // Find earliest ledger sequence stored
+        std::tie(ledger, seq, hash) = loadLedgerHelper(
+            "WHERE LedgerSeq >= " + std::to_string(earliestSeq()) +
+            " order by LedgerSeq asc limit 1", app_);
+        if (!ledger)
+        {
+            JLOG(j_.error()) <<
+                "No ledgers found to import";
+            return;
+        }
+        earliestIndex = seqToShardIndex(seq);
+        if (seq != firstLedgerSeq(earliestIndex))
```

Comment:
While diving into this code, I found an oddity in the assert and subsequent line at https://github.com/ripple/rippled/blob/cd08ceba034ff6fac1c5db7cda36c6a435a63890/src/ripple/nodestore/impl/DatabaseShardImp.h#L88-L89

I'm not clear on how what if clause is checking.

Reply:
The first ledger of a truncated shard. eg. XRP Genesis or alt using some other earliest truncated shard.